common/defs.go

```diff
@@ -21,10 +21,19 @@ import (
 	"fmt"
 	"net"
 	"strings"
+	"encoding/json"
 )
 
 // Here we only keep type definitions and struct definitions with no behavior.
 
+func String(i interface{}) string {
```

Comment:
Well, the reason it produces JSON is that JSON is really easy to read (vs reading %v or %#v), and really easy to create. Compare logs where %#v is used vs JSON and you will see what is easier.